File: 1. 两数之和/main.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        if len(nums) == 0:
            return []
        if len(nums) == 1:
            if nums[0] == target:
                return [0]
            else:
                return []
        if len(nums) == 2:
            if nums[0] + nums[1] == target:
                return [0, 1]
            else:
                return []
        for i in range(0, len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []

solution = Solution()
result = solution.twoSum([3,2,4], 6)
print(result)

File: 1. 两数之和/rust/main.rs
struct Solution {

}

use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hash: HashMap<i32, i32> = HashMap::new();
        let length = nums.len();

        for i in 0..length {
            let rest = nums[i];
            if hash.contains_key(&rest) {
                let index = hash[&rest];
                return vec![index, i as i32];
            } else {
                hash.insert(target - rest, i as i32);
            }
        }
        return vec![];
    }
}

fn main() {
    let result = Solution::two_sum(vec![3,3], 6);
    println!("{:?}", result);
}